wasm/wat/convert.go

```diff
@@ -0,0 +1,83 @@
+package wat
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/tetratelabs/wazero/wasm"
+)
+
+// TextToBinary parses the configured source into a wasm.Module. This function returns when the source is exhausted or
+// an error occurs.
+//
+// Here's a description of the return values:
+// * result is the module parsed or nil on error
+// * err is a FormatError invoking the parser, dangling block comments or unexpected characters.
+func TextToBinary(source []byte) (result *wasm.Module, err error) {
+	// First, attempt to parse the module into a basic structure representing the text format. If this errs, return
+	// immediately without wrapping parseModule returns FormatError, which is pre-formatted.
+	var m *module
+	if m, err = parseModule(source); err != nil {
```

Comment:
why not just ?
```suggestion
	m, err = parseModule(source)
	if err != nil {
```

Reply:
because that adds an IDE warning about re-using the name `err` from the return type

Reply:
sg!